### ISSUE TITLE
New note gets added to parent, grandparent... etc. folders in the new nested tag system. 

### ISSUE THREAD

**Issue by makarandh** [standardnotes/desktop#831]
**Repro steps:**

1. Created nested tags/folder if not already present. E.g. Create a folder Grandparent. Inside it create another tag/folder called Parent. Inside it create a folder called Child. 
2. In the child folder create a new note. 

**Expected Behavior:**

The new note should appear only under Child folder but not under Parent and Grandparent folder. The only tag it should have is Grandparent/Parent/Child

**Actual Behavior **

The new note appears in all 3: Child, Parent, Grandparent. It has 3 tags: Grandparent, Grandparent/Parent, Grandparent/Parent/Child

The new note appears 

**Environment:**

Standard Notes 3.10.0
Debian 11 (Bullseye)

**Comment by moughxyz** [comment]
Why wouldn't you want child notes to appear when navigating parent tag? Can you give an example?

For the record parent tags are added explicitly to the note and you can remove them if you don't want this behavior.

**Comment by makarandh** [comment]
This is a change in behavior from how folders used to work. 

Say I have a parent folder called Python where I have Python related notes. 
I make a child folder in Python called Flask where I put exclusively Flask related notes.

When I click on Python, the parent folder, I dont' want to see Flask related notes for obvious reason. I just want to see my python notes.

**Comment by makarandh** [comment]
I would like to add: 

If this change in behavior of your app is planned (and not an unintended side effect), then it would be nice if you could add a setting such that those of us who want the old behavior could set it that way. Having to delete tags, especially when there are many levels deep nesting, having to delete multiple tags every time I create a new note is a hassle.

**Comment by jceb** [comment]
I think this is the same issue as https://github.com/standardnotes/web/issues/597 that I raised last year in summer. Since then, the issue had been fixed and opened again.
Like @makarandh I'd like to see a setting to make this configurable. The current behavior is hardly bearable.

**Comment by codeoholic** [comment]
I think of this as a good feature.
Taking @makarandh's example here. Flask is an extension of Python making everything in Flask folder part of Python. If there are Python notes which cannot be categorized further then they probably need more work to categorize them into something granular.

**Comment by moughxyz** [comment]
But let's say you have a folder Contacts. Then inside you have Personal and Business. If you wanted to view all Contacts, personal or business, shouldn't you be able to click the top level Contacts to view them?

**Comment by codeoholic** [comment]
Yes, I agree to this logic @moughxyz. Any contact that is not categorized anywhere in Personal or Professional, needs another category and definitely don't be the only item when Contacts folder is clicked.

**Comment by makarandh** [comment]
> 
> Note that this isn't a change of behavior from the Folders extension. It also added parent tags recursively.
> 

If I memory serves me right, this is only partially correct. 

Previous behavior: 
For a 3 level deep note, tag that will be added: grandparent.parent.child
So a total of one tag separated by dots. 

Current behavior:
For the same note tags will be added: grandparent/parent/child grandparent/parent grandparent
So a total of 3 tags, each separated by slash (/) where applicable.

**Comment by atmoio** [comment]
Someone just tweeted this at us:

<img width="579" alt="Screen Shot 2022-02-16 at 7 26 33 AM" src="https://user-images.githubusercontent.com/3277844/154274136-a0560205-8894-419b-ba6f-0a10f1d5d259.png">

Told you there are two camps of people 😂

**Comment by rubonabo** [comment]
I totally agree with makarandh!!!
The new behavior is a REGRESSION that is not acceptable for me!
I have several accounts. One of my accounts has 2913 notes and 1791 tags.
I constantly improve my folders structure and I have only notes in the child folders.
My way of working was efficient, clear!
The new behavior creates a mess I am not able to maintain with the risk to delete notes when I try to clean the upper levels.
I accept that people prefer a different way of working but I have been working with SN for 3 years.
It is not acceptable to change suddenly the initial behavior: this is called a REGRESSION!!!
SN is my daily tool for my knowledge base

**Comment by moughxyz** [comment]
We'll probably just have to make this a user preference setting. No other way to do it. We'll add it to our roadmap.

**Comment by cyb3rz3us** [comment]
_"Told you there are two camps of people"_ - Don't think that is being debated as this really isn't a right or wrong kind of thing.  But as with the removal of Extension browsing, making changes to long-standing features\functionality with little-to-no notice + no solicitation of user feedback is generally a bad idea.
